tests/test_model_saving.py

```diff
@@ -104,6 +104,36 @@ def test_functional_model_saving():
     assert_allclose(out, out2, atol=1e-05)
 
 
+@keras_test
+def test_model_saving_to_stream():
+    inputs = Input(shape=(3,))
+    x = Dense(2)(inputs)
+    outputs = Dense(3)(x)
+
+    model = Model(inputs, outputs)
+    model.compile(loss=losses.MSE,
+                  optimizer=optimizers.Adam(),
+                  metrics=[metrics.categorical_accuracy])
+    x = np.random.random((1, 3))
+    y = np.random.random((1, 3))
+    model.train_on_batch(x, y)
+
+    out = model.predict(x)
+    _, fname = tempfile.mkstemp('.h5')
+    with h5py.File(fname, mode='w') as h5file:
```

Comment:
Shouldn't this be `mode='a'` to allow both reading and writing? I think the file is being opened in write-only mode right now.